docs/how_backtrader_executes_code.rst

```diff
@@ -0,0 +1,105 @@
+TLDR: How BackTrader Executes Code
+==================================
+
+Preonce, Once, Next, Init – Why Isn’t My Code Working?
+------------------------------------------------------
+
+Backtrader evolved from a backtesting only platform to a platform that performs real-time
+processing of data. It now has signal generation from strategies, optimization and trade
+execution. Because of this there are different ways for Backtrader to execute indicator code
+varying depending on whether or not the data is live or historical and preloaded.
+
+Coding for Backtesting:
+
+This is how Backtrader processes indicator code when backtesting preloaded data.
+
+By LINES:
+
+	Init() - processes bars as lines, all at once
```

Comment:
Trying to pretend as a casual framework user,  just to get into the context:

* "Init()"  -  do you mean `__init__` method ?  or Init() phase ?  Init() of indicator, strategy, data feed?
* "processes bars as lines, all at once"  -  What is the difference between processing bars and processing lines?  AFAIU even preloaded data is processed bar by bar and not at once nor parallel nor vectorized (unless I'm missing something).  So probably this needs to be either described in more detail or some background information or reference to the existing doc to be added before this paragraph.

Reply:
Sorry I missed your comments before merging.  

1) The init() phase of the indicator is meant here. 

2) I'm not trying to go into the internals that backtrader iterates through bars even in the lines operations, just give an overview that next() process "by bars" meaning step-by-step rather than as lines operations. It's assumed in this document that the reader knows the difference between the two as they are covered in the existing docs.

Reply:
The 'lines operation' is not defined well I'm afraid